sycl/test/usm/allocatorll.cpp

```diff
@@ -0,0 +1,94 @@
+// RUN: %clang -std=c++11 -fsycl %s -o %t1.out -lstdc++ -lOpenCL -lsycl
+// RUN: %CPU_RUN_PLACEHOLDER %t1.out
+//==---- allocatorll.cpp - Device Memory Linked List Allocator test --------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <CL/sycl.hpp>
+
+using namespace cl::sycl;
+
+int numNodes = 4;
+
+struct Node {
+  Node() : pNext(nullptr), Num(0xDEADBEEF) {}
+
+  Node *pNext;
+  uint32_t Num;
+};
+
+class foo;
+int main() {
+  bool failed = false;
+
+  queue q;
+  auto dev = q.get_device();
+  auto ctxt = q.get_context();
+
+  usm_allocator<Node, usm::alloc::device> alloc(&ctxt, &dev);
+
+  Node *d_head = nullptr;
+  Node *d_cur = nullptr;
+  Node h_cur;
+
+  for (int i = 0; i < numNodes; i++) {
+    if (i == 0) {
+      d_head = alloc.allocate(1);
+      if (d_head == nullptr) {
```

Comment:
`allocate` can't return `nullptr` - it throws an exception.

Reply:
Good catch.